src/nfts/NftGallery.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { RefreshControl, ScrollView, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { NftEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import Touchable from 'src/components/Touchable'
+import ImageErrorIcon from 'src/icons/ImageErrorIcon'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import NftImage from 'src/nfts/NftImage'
+import NftsLoadError from 'src/nfts/NftsLoadError'
+import { nftsErrorSelector, nftsLoadingSelector, nftsSelector } from 'src/nfts/selectors'
+import { fetchNfts } from 'src/nfts/slice'
+import { NftOrigin } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+
+// This is used instead of a gap between the images because the gap is not supported in RN yet.
+const imageSize = (variables.width - Spacing.Regular16 * 3) / 2
+
+export default function NftGallery() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const loading = useSelector(nftsLoadingSelector)
+  const error = useSelector(nftsErrorSelector)
+  const nfts = useSelector(nftsSelector)
+
+  function onRefresh() {
+    dispatch(fetchNfts())
+  }
+
+  useEffect(() => ValoraAnalytics.track(NftEvents.nft_gallery_screen_open), [])
+
+  return (
+    <SafeAreaView testID="NftGallery" style={styles.safeAreaContainer} edges={['top']}>
+      <DrawerTopBar middleElement={<Text>{t('nftGallery.title')}</Text>} />
+      <ScrollView
```

Comment:
could we use Flatlist instead for better performance, since the list of nft's could be quite long? there is a `numColumns` prop that you could use i think?

Reply:
Done in 1bbfdf6a67484f3d1145ade2c32cc2d20538c517!